docs/getting_started/integration_tests.md

```diff
@@ -0,0 +1,150 @@
+
+[embedmd]:# (../../pkg/test/suite/suite.go /type TestingSuite interface/ /\n}/)
+```go
+type TestingSuite interface {
+	Env() *env.Environment
+	SetEnv(environment *env.Environment)
+	SetT(t *testing.T)
+	T() *testing.T
+	SetupSuite() []Option
+}
+```
+
+The starting point for a Gosoline integration test execution is the `suite.Run` function:
+
+[embedmd]:# (../../pkg/test/suite/run.go /func Run/ /\)/)
+```go
+func Run(t *testing.T, suite TestingSuite, extraOptions ...Option)
+```
+
+This method will use reflection to find all test cases declared by a given _TestingSuite_ object, apply the Option then create a kernel with whatever modules or APIs were declared in the `SetupSuite` and `SetupApiDefinitions`, and lastly run the tests.
+
+Each Gosoline integration test follows the same format:
+
+- Creates an object which implements _TestingSuite_
+- Implements the `SetupSuite` method for that object
+- Has at least one `Test...` method
+- It calls `suite.Run`
+
+## Practical example
+
+Inside `examples/getting_started/integration` we have an integration test for _money-exchange_. It is comprised out of two files, `api_test.go` and `fixtures.go`.
+
+`fixtures.go` contains initial values for the `currency` key value store. Its most important part is:
+
+[embedmd]:# (../../examples/getting_started/integration/fixtures.go /var fixtureSets/ /\n}/)
+```go
+var fixtureSets = []*gosoFixtures.FixtureSet{
+	{
+		Enabled: true,
+		Writer:  gosoFixtures.ConfigurableKvStoreFixtureWriterFactory("currency"),
+		Fixtures: []interface{}{
+			&gosoFixtures.KvStoreFixture{
+				Key:   "GBP",
+				Value: 1.23,
+			},
+			&gosoFixtures.KvStoreFixture{
+				Key:   "2021-01-03-GBP",
+				Value: 1.24,
+			},
+		},
+	},
+}
+```
+
+This _FixtureSet_ holds exchange rate data for GBP, which will be needed by the test.
+
+`api_test.go` declares _ApiTestSuite_ which will implement _TestingSuite_ :
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /type ApiTestSuite/ /\n}/)
+```go
+type ApiTestSuite struct {
+	suite.Suite
+
+	client *resty.Client
+	clock  clock.Clock
+}
+```
+
+It also declares a normal unit test which makes use of the above struct, and calling _suite.Run_:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func TestApiTestSuite/ /\n}/)
+```go
+func TestApiTestSuite(t *testing.T) {
+	suite.Run(t, &ApiTestSuite{
+		clock: clock.NewFakeClockAt(time.Now().UTC()),
+	})
+}
+```
+
+Another important thing about _ApiTestSuite_ is how it configures the kernel which will run the tests:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) SetupSuite/ /\n}/)
+```go
+func (s *ApiTestSuite) SetupSuite() []suite.Option {
+	clock.WithUseUTC(true)
+	clock.Provider = s.clock
+
+	return []suite.Option{
+		suite.WithLogLevel("info"),
+		suite.WithConfigFile("../api/config.dist.yml"),
+		suite.WithFixtures(fixtureSets),
+		suite.WithClockProvider(s.clock),
```

Comment:
You already set the clock.Provider, why are you doing it again?

Reply:
Why is that? Why is the test so sensitive to the current time?

Reply:
Ah, I see. Because of this in `gosoline/pkg/test/suite/run.go:79`:
```
	setupOptions := []Option{
		WithClockProvider(clock.NewFakeClock()),
	}
```

Well, we could also just use a real clock, then the test would work... actually not really sure why we by default run tests with a fake clock